docs/development/integration_tests.md

```diff
@@ -1,44 +1,56 @@
-  - (optional) If you want to use a machineClass of your own for the test
-    - Deploy a secret named `test-mc-secret` that contains the provider secret and cloud-config into the kubernetes cluster. Refer [these](https://github.com/gardener/machine-controller-manager/tree/master/kubernetes/machine_classes) machineClass templates for the same. Make sure that value of `metadata.namespace` is the technicalID i.e `shoot--<project>--<shoot-name>` of the shoot.
-    - Create a `dev/machineclassv1.yaml` file.
-      -  The value of `providerSpec.secretRef.name` should be the secret created in the previous step. 
-      -  The value of `providerSpec.secretRef.namespace` should be technical-ID(`shoot--<project>--<shoot-name>`) of the shoot 
-      -  The name of the machineclass itself should be `test-mc-v1`. 
-      -  The value of `metadata.namespace`  should be technicalID of the shoot.
-  
-- If none of the images or only one image is specified, `machine-controller-manager` repository will be cloned automatically. Incase, this repository already exists in local system, then create a softlink as below which helps to test changes in `machine-controller-manager` quickly.
+1. Clone the repository `machine-controller-manager-provider-{provider-name}` on the local system.
+1. Navigate to `machine-controller-manager-provider-{provider-name}` directory and create a `dev` sub-directory in it.
+1. Copy the kubeconfig of Control Cluster  from into `dev/control-kubeconfig.yaml`. 
+1. (optional) Copy the kubeconfig of Target Cluster  into `dev/target-kubeconfig.yaml` and update the `Makefile` variable `TARGET_KUBECONFIG` to point to `dev/target-kubeconfig.yaml`.
+1. If the tags on instances & associated resources on the provider are of `String` type (for example, GCP tags on its instances are of type `String` and not key-value pair) then add `TAGS_ARE_STRINGS := true` in the `Makefile`.
+1. Atleast, one of the two controllers' container images must be set in the `Makefile` variables `MCM_IMAGE_TAG` and `MC_IMAGE_TAG` for the controllers to run in the Control Cluster . These images will be used along with `kubernetes/deployment.yaml` to deploy/update controllers in the Control Cluster . If the intention is to run the controllers locally then unset the variables `MCM_IMAGE_TAG` and `MC_IMAGE_TAG` and set variable `MACHINE_CONTROLLER_MANAGER_DEPLOYMENT_NAME := machine-controller-manager` in the `Makefile`.
+7. If none of the images are specified, `machine-controller-manager` repository will be cloned automatically. Incase, this repository already exists in local system, then create a softlink as below which helps to test changes in `machine-controller-manager` quickly.
     ```bash
     ln -sf <path-for-machine-controller-manager-repo> dev/mcm
     ```
-- There is a rule `test-integration` in the `Makefile`, which can be used to start the integration test:
+## Scenario based additional configurations
```

Comment:
It is better to have sections for easier consumption.

Reply:
Yep this way its really easy to understand the flow. Thanks